### PR TITLE
load_realtime: a daemon to init kraken with fresh data

### PR DESCRIPTION
**Pull request by TeXitoi**
Jira: http://jira.canaltp.fr/browse/NAVP-313

### CODE REVIEW

File: kirin/command/load_realtime.py~ [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2001-2015, Canal TP and/or its affiliates. All rights reserved.
```
Comment (xlqian) [inline kirin/command/load_realtime.py~]: ` load_realtime.py~` ??

File: kirin/core/model.py [hunk 2]
```diff
@@ -196,11 +196,12 @@ def __repr__(self):
         return '<RealTimeUpdate %r>' % self.id
 
     @classmethod
-    def all(cls, contributors, start_date, end_date=None):
+    def all(cls, contributors, start_date=None, end_date=None):
```
Comment (xlqian) [inline kirin/core/model.py]: Personally, I found this function is confusing, what if let's say "start_date" and "end_date" are all not None? 

maybe:
`def all(cls, contributors, date, is_preceded_by =True)`?

**Comment by TeXitoi** [comment]
Jenkins test this please.

File: kirin/core/populate_pb.py [hunk 3]
```diff
@@ -43,15 +43,16 @@ def to_posix_time(date_time):
...
 
     feed.header.incrementality = gtfs_realtime_pb2.FeedHeader.DIFFERENTIAL
     feed.header.gtfs_realtime_version = '1'
     feed.header.timestamp = to_posix_time(datetime.datetime.utcnow())
 
-    for trip_update in real_time_update.trip_updates:
-        fill_entity(feed.entity.add(), trip_update)
+    for real_time_update in real_time_updates:
+        for trip_update in real_time_update.trip_updates:
+            fill_entity(feed.entity.add(), trip_update)
```
Comment (xlqian) [inline kirin/core/populate_pb.py]: `[fill_entity(feed.entity.add(), trip_update) for real_time_update in real_time_updates for trip_update in real_time_update ]`

File: kirin/command/load_realtime.py~ (hunk 1 above)
Comment (TeXitoi) [inline kirin/command/load_realtime.py~]: oups

**Comment by TeXitoi** [comment]
Jenkins, test this please

File: kirin/core/populate_pb.py (hunk 3 above)
Comment (TeXitoi) [inline kirin/core/populate_pb.py]: Using a list comprehension only for side effect seems strange to me.

**Comment by TeXitoi** [comment]
Jenkins, test this please

File: navitia-proto [hunk 4]
```diff
@@ -0,0 +1 @@
+Subproject commit dbb0516153c314e309569f8a265d8747d4df3827
```
Comment (pbougue) [inline navitia-proto]: update sha-1

File: kirin/rabbitmq_handler.py [hunk 5]
```diff
@@ -59,6 +64,46 @@ def info(self):
...
+            log.info('getting a request: {}'.format(task))
+            if task.action != task_pb2.LOAD_REALTIME or not task.load_realtime:
+                return
+
+            feed = convert_to_gtfsrt(RealTimeUpdate.all(task.load_realtime.contributors))
+
+            with self._get_producer() as producer:
+                producer.publish(feed.SerializeToString(), routing_key=task.load_realtime.queue_name)
+
+        route = 'task.load_realtime.INSTANCE'
```
Comment (kinnou02) [inline kirin/rabbitmq_handler.py]: prefer "task.load_realtime.*"

File: kirin/rabbitmq_handler.py [hunk 6]
```diff
@@ -59,6 +64,46 @@ def info(self):
...
+                return
+
+            log.info('getting a request: {}'.format(task))
+            if task.action != task_pb2.LOAD_REALTIME or not task.load_realtime:
+                return
+
+            feed = convert_to_gtfsrt(RealTimeUpdate.all(task.load_realtime.contributors))
+
+            with self._get_producer() as producer:
+                producer.publish(feed.SerializeToString(), routing_key=task.load_realtime.queue_name)
```
Comment (kinnou02) [inline kirin/rabbitmq_handler.py]: ```routing_key``` directly define the queue to use when no exchange if defined?

File: navitia-proto (hunk 4 above)
Comment (TeXitoi) [inline navitia-proto]: That's not an update, I add navitia-proto as a submodule (because task.proto is used by kraken to make its request).

Comment (TeXitoi) [inline navitia-proto]: Or you want I update navitia-proto up to current master?

File: kirin/rabbitmq_handler.py (hunk 6 above)
Comment (TeXitoi) [inline kirin/rabbitmq_handler.py]: Yep, without an exchange, it's the name of the queue. https://www.rabbitmq.com/tutorials/tutorial-one-python.html

File: kirin/core/populate_pb.py (hunk 3 above)
Comment (antoine-de) [inline kirin/core/populate_pb.py]: I tend to aggry with @TeXitoi on this

File: navitia-proto (hunk 4 above)
Comment (pbougue) [inline navitia-proto]: Yep :)

**Comment by TeXitoi** [comment]
For the tests, I've added a ticket as testing is lower priority than a totally working removal of vj.

http://jira.canaltp.fr/browse/NAVP-338

OK to merge?

**Pull request merged**